src/components/add-boiler-house-form/add-boiler-house-form.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useEffect } from "react";
+
+import { useBoilerHouseOptions } from "./hooks/useBoilerHouseOptions";
+import { useSearchForBoilerHouse } from "./hooks/useSearchForBoilerHouse";
+
+import { Asset } from "@mtfh/common/lib/api/asset/v1";
+import { Center, Spinner } from "@mtfh/common/lib/components";
+
+interface Props {
+  setShowSuccess: React.Dispatch<React.SetStateAction<boolean>>;
+  setRequestError: React.Dispatch<React.SetStateAction<string | null>>;
+  asset: Asset;
+}
+
+export const AddBoilerHouseForm = ({ setShowSuccess, setRequestError, asset }: Props) => {
+  const {
+    boilerHouseOption,
+    setBoilerHouseOption,
+    boilerHouseOptionError,
+    handleSubmit,
+    resetForm: resetSelectBoilerHouseForm,
+  } = useBoilerHouseOptions(asset, setShowSuccess, setRequestError);
+
+  const {
+    searchQuery,
+    setSearchQuery,
+    searchQueryError,
+    handleSearch,
+    touched,
+    error,
+    loading,
+    total,
+    searchResultsData,
+  } = useSearchForBoilerHouse(resetSelectBoilerHouseForm);
+
+  useEffect(() => {
+    setRequestError(error);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [error]);
+
+  return (
+    <div>
+      <form onSubmit={handleSearch}>
+        <div className="govuk-form-group">
+          <label className="govuk-label lbh-label" htmlFor="searchQuery">
+            Search for a boiler house
+          </label>
+          <span
+            className="govuk-error-message lbh-error-message"
+            data-testid="boiler-house-search-error"
+          >
+            <span className="govuk-visually-hidden">Error: </span>
+            {searchQueryError}
+          </span>
+          <input
+            type="text"
+            className="govuk-input lbh-input"
+            name="searchQuery"
+            value={searchQuery}
+            data-testid="boiler-house-search-input"
+            onInput={(e: React.ChangeEvent<HTMLInputElement>) =>
+              setSearchQuery(e.target.value)
+            }
+          />
+          <button
+            className="govuk-button lbh-button"
+            data-testid="boiler-house-search-submit"
+            type="submit"
+          >
+            Search
+          </button>
+        </div>
+      </form>
+
+      {loading ? (
+        <Center>
+          <Spinner />
+        </Center>
+      ) : (
+        <>
+          {touched && (
```

Comment:
We have a component called InlineSearchForm which is also used in New Asset form and I believe it has a similar function. Can that be used here maybe?

Reply:
I forgot about that component. Im not sure if its worth changing now